tests/unit/mlir_lit_tests/model-import/EnumType.yml

```diff
@@ -0,0 +1,148 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+# RUN: %revngcliftopt /dev/null -m %s --import-model-types | FileCheck %s
+
+Types:
+  - ID: 104
+    Kind: PrimitiveType
+    PrimitiveKind: Signed
+    Size: 4
+  - ID: 108
+    Kind: PrimitiveType
+    PrimitiveKind: Signed
+    Size: 8
+  - ID: 204
+    Kind: PrimitiveType
+    PrimitiveKind: Unsigned
+    Size: 4
+  - ID: 208
+    Kind: PrimitiveType
+    PrimitiveKind: Unsigned
+    Size: 8
+
+    # CHECK: #my_enum1_ = #clift.enum<
+    # CHECK:   id = 1104
+    # CHECK:   name = "my_enum1"
+    # CHECK:   underlying_type = !int32_t
+    # CHECK:   fields = [
+    # CHECK:     <
+    # CHECK:       raw_value = 0
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 1
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 2147483647
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 18446744071562067968
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 18446744073709551615
+    # CHECK:     >
+    # CHECK:   ]
+    # CHECK: >
+  - ID: 1104
+    Kind: EnumType
+    CustomName: "my_enum1"
+    UnderlyingType:
+      UnqualifiedType: "/Types/104-PrimitiveType"
+    Entries:
+      - Value: 0
+      - Value: 1
+      - Value: 0x000000007fffffff # 2^31 - 1
+      - Value: 0xffffffff80000000 # 2^64 - 2^31
+      - Value: 0xffffffffffffffff # 2^64 - 1
+
+    # CHECK: #my_enum2_ = #clift.enum<
+    # CHECK:   id = 1108
+    # CHECK:   name = "my_enum2"
+    # CHECK:   underlying_type = !int64_t
+    # CHECK:   fields = [
+    # CHECK:     <
+    # CHECK:       raw_value = 0
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 1
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 9223372036854775807
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 9223372036854775808
+    # CHECK:     >
+    # CHECK:     <
+    # CHECK:       raw_value = 18446744073709551615
+    # CHECK:     >
```

Comment:
Lots of newlines, I wonder if this was intentional

Reply:
We do want to ignore other things such as enumerator name between the angle brackets. That's why I used multiple lines. Perhaps there is a way to do this in a single line using FileCheck, but I'm not sure.

Reply:
Can you paste the full output so I can get a better idea?

Reply:
```CPP
!int32_t = !clift.primitive<SignedKind 4>
!int64_t = !clift.primitive<SignedKind 8>
!uint32_t = !clift.primitive<UnsignedKind 4>
!uint64_t = !clift.primitive<UnsignedKind 8>
#my_enum1_ = #clift.enum<id = 1104, name = "my_enum1", underlying_type = !int32_t, fields = [<name = "_enum_entry_my_enum1_0", raw_value = 0>, <name = "_enum_entry_my_enum1_1", raw_value = 1>, <name = "_enum_entry_my_enum1_2147483647", raw_value = 2147483647>, <name = "_enum_entry_my_enum1_18446744071562067968", raw_value = 18446744071562067968>, <name = "_enum_entry_my_enum1_18446744073709551615", raw_value = 18446744073709551615>]>
#my_enum2_ = #clift.enum<id = 1108, name = "my_enum2", underlying_type = !int64_t, fields = [<name = "_enum_entry_my_enum2_0", raw_value = 0>, <name = "_enum_entry_my_enum2_1", raw_value = 1>, <name = "_enum_entry_my_enum2_9223372036854775807", raw_value = 9223372036854775807>, <name = "_enum_entry_my_enum2_9223372036854775808", raw_value = 9223372036854775808>, <name = "_enum_entry_my_enum2_18446744073709551615", raw_value = 18446744073709551615>]>
#my_enum3_ = #clift.enum<id = 1204, name = "my_enum3", underlying_type = !uint32_t, fields = [<name = "_enum_entry_my_enum3_0", raw_value = 0>, <name = "_enum_entry_my_enum3_1", raw_value = 1>, <name = "_enum_entry_my_enum3_4294967295", raw_value = 4294967295>]>
#my_enum4_ = #clift.enum<id = 1208, name = "my_enum4", underlying_type = !uint64_t, fields = [<name = "_enum_entry_my_enum4_0", raw_value = 0>, <name = "_enum_entry_my_enum4_1", raw_value = 1>, <name = "_enum_entry_my_enum4_18446744073709551615", raw_value = 18446744073709551615>]>
!my_enum1_1 = !clift.defined<#my_enum1_>
!my_enum2_1 = !clift.defined<#my_enum2_>
!my_enum3_1 = !clift.defined<#my_enum3_>
!my_enum4_1 = !clift.defined<#my_enum4_>
module {
  %0 = clift.undef !int32_t
  %1 = clift.undef !int64_t
  %2 = clift.undef !uint32_t
  %3 = clift.undef !uint64_t
  %4 = clift.undef !my_enum1_1
  %5 = clift.undef !my_enum2_1
  %6 = clift.undef !my_enum3_1
  %7 = clift.undef !my_enum4_1
}
```

This is what is being matched: `<name = "_enum_entry_my_enum1_0", raw_value = 0>`

Reply:
Aaaah, so `CHECK` does not necessarily imply a new line, didn't know that.

The fact that [`CHECK-SAME`](https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-same-directive) exists made me think `CHECK` implied a new line 🤔 

Anyway, it'd be nice to have a more readable output for types where the type definition is split on multiple lines, but this works for now.

If you want to reduce the number of `CHECK` you can use [regex](https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-regex-matching-syntax) and match multiple things in a single shot.

Reply:
Yeah. I figure this is more readable than regexes though, even if it gets split up on multiple lines.